In [1]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
from sqlalchemy import create_engine
import psycopg2

In [2]:
import psycopg2

connection_url = "postgresql://postgres:root@localhost:5432/ecommerce"

conn = psycopg2.connect(connection_url)
cursor = conn.cursor()

query = "SELECT * FROM products"

cursor.execute(query)

rows = []
for item in cursor:
    rows.append(item)

conn.close()

products_df = pd.DataFrame.from_records(rows, columns=['codigo', 'produto', 'preco'])
products_df.head()

codigo                                       produto    preco
0  BS001873                         Apple Lightning Cable   $21.99
1  BS001874                     SanDisk Extreme microSDXC   $31.50
2  BS001875            Seagate 2TB Expansion Portable HDD   $79.00
3  BS001876  TicWatch Pro 3 Ultra GPS Smartwatch Qualcomm  $399.00
4  BS001877                    Webcam HD 1080p Web Camera   $59.00

In [3]:
conn = psycopg2.connect(connection_url)
cursor = conn.cursor()

query = "SELECT * FROM sales"

cursor.execute(query)

rows = []
for item in cursor:
    rows.append(item)

conn.close()

sales_df = pd.DataFrame.from_records(rows, columns=['order_id', 'date', 'ship_service_level','style','codigo','courier_status','qty','ship_country','fulfillment'])
sales_df.head()

order_id       date ship_service_level    style    codigo  \
0  171_8536041_9190708  6/29/2022          Expedited   SET400  BS001879   
1  407_4176879_6571509  6/29/2022          Expedited  JNE3486  BS001890   
2  404_4382011_4681149  6/29/2022          Expedited  JNE3887  BS001873   
3  406_3412504_2471555  6/29/2022          Expedited  JNE3487  BS001888   
4  408_2481787_8248368  6/29/2022          Expedited  JNE3566  BS001898   

  courier_status qty ship_country fulfillment  
0      Cancelled   4           AU         Yes  
1      Cancelled   2           BR         Yes  
2      Cancelled   2           BR         Yes  
3      Cancelled   2           BR         Yes  
4      Cancelled   2           BR         Yes

### Questions
Based on the data above, the following questions were asked:
z


] and less

### Views created for each question

In [4]:
# What's the relation between price and quantity purchased as a whole?
conn = psycopg2.connect(connection_url)
cursor = conn.cursor()
query = "SELECT * FROM price_quantity_relation"

cursor.execute(query)

rows = []
for item in cursor:
    rows.append(item)

conn.close()

price_quantity_relation_df = pd.DataFrame.from_records(rows, columns=['preco','total_quantity_purchased'])
price_quantity_relation_df.head()

preco total_quantity_purchased
0   $14.00                    12449
1   $14.99                    15247
2  $149.00                    12332
3  $179.00                    12324
4   $21.99                    25018

In [67]:
price_quantity_correlation = round(pd.to_numeric(price_quantity_relation_df['preco'].str[1:]).corr(pd.to_numeric(price_quantity_relation_df['total_quantity_purchased'])),4)
print(f"Price-quantity correlation: {price_quantity_correlation}")

Price-quantity correlation: -0.1049


Given the price-quantity correlation of -0.1049, it's possible to conclude that although a higher product price leads to a smaller demand, it is a weak correlation.

In [5]:
#Did products with smaller prices have larger or smaller revenue?
conn = psycopg2.connect(connection_url)
cursor = conn.cursor()
query = "SELECT * FROM price_revenue_comparison"

cursor.execute(query)

rows = []
for item in cursor:
    rows.append(item)

conn.close()

price_revenue_comparison_df = pd.DataFrame.from_records(rows, columns=['preco','total_revenue'])
price_revenue_comparison_df.head()

preco total_revenue
0   $14.00     174286.00
1   $14.99     228552.53
2  $149.00    1837468.00
3  $179.00    2205996.00
4   $21.99     550145.82

In [68]:
price_revenue_correlation = round(pd.to_numeric(price_revenue_comparison_df['preco'].str[1:]).corr(pd.to_numeric(price_revenue_comparison_df['total_revenue'])),4)
print(f"Price-revenue correlation: {price_revenue_correlation}")

Price-revenue correlation: 0.9839


Given the price-revenue correlation of 0.9839, it's possible to conclude that there is a very strong positive correlation between price and revenue of a given product.

In [80]:
# What products had more and less orders cancelled?
conn = psycopg2.connect(connection_url)
cursor = conn.cursor()
query = "SELECT * FROM cancelled_orders"

cursor.execute(query)

rows = []
for item in cursor:
    rows.append(item)

conn.close()

cancelled_orders_df = pd.DataFrame.from_records(rows, columns=['codigo','produto','cancelled_orders_count'])
cancelled_orders_df.head()

codigo                                            produto  \
0  BS001882  HEYMIX Powerboard, Power Strip with 8 Outlets,...   
1  BS001896                           Charmast Mini Power Bank   
2  BS001881        TP-Link Tapo Pan/Tilt Smart Security Camera   
3  BS001894  Sony WH1000XM4 Noise Canceling Wireless Headph...   
4  BS001892                            Kindle Paperwhite (8GB)   

   cancelled_orders_count  
0                     192  
1                     215  
2                     236  
3                     237  
4                     238

In [84]:
cancelled_orders_ranked_df = cancelled_orders_df.sort_values(by=['cancelled_orders_count'], ascending=False)
print(display(cancelled_orders_ranked_df[['produto','cancelled_orders_count']].head()),display(cancelled_orders_ranked_df[['produto','cancelled_orders_count']].tail()))

produto  cancelled_orders_count
23  YABER WiFi Projector Mini Portable Projector 6...                     246
25                 LG 29WL50S - 29 inch UltraWide UHD                     240
4                             Kindle Paperwhite (8GB)                     238
3   Sony WH1000XM4 Noise Canceling Wireless Headph...                     237
9   HyperX QuadCast S – RGB USB Condenser Micropho...                     237

produto  cancelled_orders_count
17                 Seagate 2TB Expansion Portable HDD                     192
0   HEYMIX Powerboard, Power Strip with 8 Outlets,...                     192
13  Fingertip Pulse Oximeter, Blood Oxygen Saturation                     191
11                         Webcam HD 1080p Web Camera                     191
8                         Samsung T37F Monitor UHD 37                     177

None None


The five products with more and less orders cancelled are displayed above. To change the amount of top and bottom products displayed, add a parameter to the head or tail method.

In [101]:
# What products were more purchased in each month?
conn = psycopg2.connect(connection_url)
cursor = conn.cursor()
query = "SELECT * FROM popular_products_per_month"

cursor.execute(query)

rows = []
for item in cursor:
    rows.append(item)

conn.close()

popular_products_per_month_df = pd.DataFrame.from_records(rows, columns=['month','codigo','produto','total_quantity'])
popular_products_per_month_df.head()

month    codigo                                       produto total_quantity
0     3  BS001873                         Apple Lightning Cable             14
1     3  BS001874                     SanDisk Extreme microSDXC             12
2     3  BS001875            Seagate 2TB Expansion Portable HDD             16
3     3  BS001876  TicWatch Pro 3 Ultra GPS Smartwatch Qualcomm             16
4     3  BS001877                    Webcam HD 1080p Web Camera             14

In [99]:
(popular_products_per_month_df.sort_values(['month','total_quantity'], ascending=[True, False]).groupby(['month'], as_index = False, sort = False).nth([0]))

month    codigo                                            produto  \
19     3  BS001892                            Kindle Paperwhite (8GB)   
45     4  BS001889               Taco Cat Goat Cheese Pizza Card Game   
75     5  BS001890                       Nintendo Switch Console Lite   
92     6  BS001878  Logitech MK270R Wireless Keyboard and Mouse Combo   

   total_quantity  
19             30  
45           5851  
75           4994  
92           4580

Displayed above are the top items purchased by month. Given that the highest demand in March was 30, we may assume that the data may not be right for that month.

In [8]:
# Was there preference for faster delivery on some products?
conn = psycopg2.connect(connection_url)
cursor = conn.cursor()
query = "SELECT * FROM expedited_delivery_preference"

cursor.execute(query)

rows = []
for item in cursor:
    rows.append(item)

conn.close()

expedited_delivery_preference_df = pd.DataFrame.from_records(rows, columns=['codigo','produto','ship_service_level','expedited_orders','total_orders_count'])
expedited_delivery_preference_df.head()

codigo                                       produto ship_service_level  \
0  BS001873                         Apple Lightning Cable          Expedited   
1  BS001874                     SanDisk Extreme microSDXC          Expedited   
2  BS001875            Seagate 2TB Expansion Portable HDD          Expedited   
3  BS001876  TicWatch Pro 3 Ultra GPS Smartwatch Qualcomm          Expedited   
4  BS001877                    Webcam HD 1080p Web Camera          Expedited   

   expedited_orders  total_orders_count  
0              2790                4131  
1              2825                4131  
2              2835                4131  
3              2860                4131  
4              2861                4130

In [152]:
expedited_order_percentage = round(expedited_delivery_preference_df['expedited_orders']/expedited_delivery_preference_df['total_orders_count'],2)
data_entry = {'Products': expedited_delivery_preference_df['produto'], 'Percentage of expedited orders': expedited_order_percentage}
expedited_preference_df = pd.DataFrame(data=data_entry)
expedited_preference_df.sort_values(by=['Percentage of expedited orders'],ascending=False)

Products  \
28                 LG 29WL50S - 29 inch UltraWide UHD   
9   HEYMIX Powerboard, Power Strip with 8 Outlets,...   
25                     Apple AirPods (3rd Generation)   
19                            Kindle Paperwhite (8GB)   
23                           Charmast Mini Power Bank   
17                       Nintendo Switch Console Lite   
16               Taco Cat Goat Cheese Pizza Card Game   
15    LEGO Ideas Home Alone McCallisters’ House 21330   
24                            New Apple AirTag 4 Pack   
13  Withings BPM Connect Wireless Blood Pressure M...   
12  Fingertip Pulse Oximeter, Blood Oxygen Saturation   
10                                  Apple AirPods Pro   
22                   Google Chromecast 3rd Generation   
8         TP-Link Tapo Pan/Tilt Smart Security Camera   
26                              Apple MagSafe Charger   
6                         Samsung T37F Monitor UHD 37   
4                          Webcam HD 1080p Web Camera   
3        TicWatch Pro 3 Ultra GPS Smartwatch Qualcomm   
2                  Seagate 2TB Expansion Portable HDD   
20    Apple Lightning to 3.5mm Headphone Jack Adapter   
27  YABER WiFi Projector Mini Portable Projector 6...   
0                               Apple Lightning Cable   
21  Sony WH1000XM4 Noise Canceling Wireless Headph...   
18    PHILIPS Sonicare 9900 Prestige Power Toothbrush   
1                           SanDisk Extreme microSDXC   
11  Digital Blood Pressure Monitor with Large Colo...   
7   HyperX QuadCast S – RGB USB Condenser Micropho...   
5   Logitech MK270R Wireless Keyboard and Mouse Combo   
14         LEGO Ideas 21319 Central Perk Building Kit   

    Percentage of expedited orders  
28                            0.70  
9                             0.70  
25                            0.69  
19                            0.69  
23                            0.69  
17                            0.69  
16                            0.69  
15                            0.69  
24                            0.69  
13                            0.69  
12                            0.69  
10                            0.69  
22                            0.69  
8                             0.69  
26                            0.69  
6                             0.69  
4                             0.69  
3                             0.69  
2                             0.69  
20                            0.69  
27                            0.68  
0                             0.68  
21                            0.68  
18                            0.68  
1                             0.68  
11                            0.68  
7                             0.68  
5                             0.68  
14                            0.68

Displayed above are the percentage of orders expedited for each product. All follow the same central tendency.

In [9]:
# What countries of origin were more common?
conn = psycopg2.connect(connection_url)
cursor = conn.cursor()
query = "SELECT * FROM common_origin_countries"

cursor.execute(query)

rows = []
for item in cursor:
    rows.append(item)

conn.close()

common_origin_countries_df = pd.DataFrame.from_records(rows, columns=['ship_country','orders_count'])
common_origin_countries_df.head()

ship_country  orders_count
0           AU         14325
1           BR         14384
2           DK         14325
3           FR         14325
4           GE         14322

Considering the five countries in the dataset, all have approximately the same number of orders, which is unusual but it is not possible to determine the root cause with just this dataset.

In [10]:
# What products had orders with larger quantities?
conn = psycopg2.connect(connection_url)
cursor = conn.cursor()
query = "SELECT * FROM products_larger_quantities"

cursor.execute(query)

rows = []
for item in cursor:
    rows.append(item)

conn.close()

products_larger_quantities_df = pd.DataFrame.from_records(rows, columns=['codigo','produto','max_quantity'])
products_larger_quantities_df.head()

codigo                                            produto max_quantity
0  BS001882  HEYMIX Powerboard, Power Strip with 8 Outlets,...            5
1  BS001896                           Charmast Mini Power Bank            5
2  BS001881        TP-Link Tapo Pan/Tilt Smart Security Camera            5
3  BS001894  Sony WH1000XM4 Noise Canceling Wireless Headph...            5
4  BS001892                            Kindle Paperwhite (8GB)            5

The highest quantity in a single order for all products was 5.

In [153]:
# What products had orders with smaller quantities?
conn = psycopg2.connect(connection_url)
cursor = conn.cursor()
query = "SELECT * FROM products_smaller_quantities"

cursor.execute(query)

rows = []
for item in cursor:
    rows.append(item)

conn.close()

products_smaller_quantities_df = pd.DataFrame.from_records(rows, columns=['codigo','produto','min_quantity'])
products_smaller_quantities_df.head()

codigo                                            produto min_quantity
0  BS001882  HEYMIX Powerboard, Power Strip with 8 Outlets,...            1
1  BS001896                           Charmast Mini Power Bank            1
2  BS001881        TP-Link Tapo Pan/Tilt Smart Security Camera            1
3  BS001894  Sony WH1000XM4 Noise Canceling Wireless Headph...            1
4  BS001892                            Kindle Paperwhite (8GB)            1

All products had a minimal quantity per order of 1, which is to be expected given that those are tech products with higher prices.

In [13]:
# Does price affect the quantity of products purchased in a single order?
conn = psycopg2.connect(connection_url)
cursor = conn.cursor()
query = "SELECT * FROM price_quantity_correlation"

cursor.execute(query)

rows = []
for item in cursor:
    rows.append(item)

conn.close()

price_quantity_correlation_df = pd.DataFrame.from_records(rows, columns=['codigo','produto','preco','avg_quantity'])
price_quantity_correlation_df.head()

codigo                                            produto    preco  \
0  BS001878  Logitech MK270R Wireless Keyboard and Mouse Combo   $38.00   
1  BS001895                   Google Chromecast 3rd Generation   $55.00   
2  BS001901                 LG 29WL50S - 29 inch UltraWide UHD  $279.00   
3  BS001897                            New Apple AirTag 4 Pack  $149.00   
4  BS001888    LEGO Ideas Home Alone McCallisters’ House 21330  $449.00   

         avg_quantity  
0  3.0231729055258467  
1  2.9782135076252723  
2  2.9897009308774015  
3  2.9852335996126846  
4  3.0077463083999032

In [154]:
price_average_qty_correlation = round(pd.to_numeric(price_quantity_correlation_df['preco'].str[1:]).corr(pd.to_numeric(price_quantity_correlation_df['avg_quantity'])),4)
print(f"Price-quantity per order correlation: {price_average_qty_correlation}")

Price-quantity per order correlation: -0.2417


Given the price-quantity per order correlation of -0.2417, it's possible to conclude that there is a somewhat weak negative correlation between price and average quantity in each order.

In [14]:
# Does a higher quantity of items purchased in a single order increase the likelihood of expedited delivery?
conn = psycopg2.connect(connection_url)
cursor = conn.cursor()
query = "SELECT * FROM quantity_expedited_delivery_correlation"

cursor.execute(query)

rows = []
for item in cursor:
    rows.append(item)

conn.close()

quantity_expedited_delivery_correlation_df = pd.DataFrame.from_records(rows, columns=['codigo','produto','avg_quantity','total_orders','expedited_orders'])
quantity_expedited_delivery_correlation_df.head()

codigo                                            produto  \
0  BS001882  HEYMIX Powerboard, Power Strip with 8 Outlets,...   
1  BS001896                           Charmast Mini Power Bank   
2  BS001881        TP-Link Tapo Pan/Tilt Smart Security Camera   
3  BS001894  Sony WH1000XM4 Noise Canceling Wireless Headph...   
4  BS001892                            Kindle Paperwhite (8GB)   

         avg_quantity  total_orders  expedited_orders  
0  2.9726458484628419          4131              2872  
1  3.0122004357298475          4590              3172  
2  3.0041592394533571          5049              3480  
3  3.0053475935828877          5049              3434  
4  2.9841521394611727          5048              3461

In [156]:
qty_expedited_correlation = round(pd.to_numeric(quantity_expedited_delivery_correlation_df['avg_quantity']).corr(pd.to_numeric(quantity_expedited_delivery_correlation_df['expedited_orders'])/quantity_expedited_delivery_correlation_df['total_orders']),4)
print(f"Correlation between quantity and demand for expedited delivery: {qty_expedited_correlation}")

Correlation between quantity and demand for expedited delivery: -0.2507


Given the -0.2507 correlation between quantity and demand for expedited delivery, it's possible to conclude there's a somewhat weak negative correlation between quantity purchased in an order and demand for expedited delivery.

In [161]:
# Which products had higher demand overall?
conn = psycopg2.connect(connection_url)
cursor = conn.cursor()
query = "SELECT * FROM high_demand_products"

cursor.execute(query)

rows = []
for item in cursor:
    rows.append(item)

conn.close()

high_demand_products_df = pd.DataFrame.from_records(rows, columns=['codigo','produto','total_quantity'])
high_demand_products_df.sort_values(by=['total_quantity'], ascending=False).head()

codigo                                            produto total_quantity
15  BS001878  Logitech MK270R Wireless Keyboard and Mouse Combo          15264
18  BS001890                       Nintendo Switch Console Lite          15256
22  BS001889               Taco Cat Goat Cheese Pizza Card Game          15247
3   BS001894  Sony WH1000XM4 Noise Canceling Wireless Headph...          15174
2   BS001881        TP-Link Tapo Pan/Tilt Smart Security Camera          15168

The dataframe above displays the 5 products with higher overall demand. To change how many products are displayed, add a parameter to the head() method.

In [157]:
# Does a higher product price increase the likelihood of requesting expedited delivery?
conn = psycopg2.connect(connection_url)
cursor = conn.cursor()
query = "SELECT * FROM price_expedited_delivery_correlation"

cursor.execute(query)

rows = []
for item in cursor:
    rows.append(item)

conn.close()

price_expedited_delivery_correlation_df = pd.DataFrame.from_records(rows, columns=['codigo','produto','preco','total_orders','expedited_orders'])
price_expedited_delivery_correlation_df.head()

codigo                                            produto    preco  \
0  BS001878  Logitech MK270R Wireless Keyboard and Mouse Combo   $38.00   
1  BS001895                   Google Chromecast 3rd Generation   $55.00   
2  BS001901                 LG 29WL50S - 29 inch UltraWide UHD  $279.00   
3  BS001897                            New Apple AirTag 4 Pack  $149.00   
4  BS001888    LEGO Ideas Home Alone McCallisters’ House 21330  $449.00   

   total_orders  expedited_orders  
0          5049              3449  
1          4131              2852  
2          5049              3513  
3          4131              2831  
4          4131              2839

In [162]:
price_expedited_correlation = round(pd.to_numeric(price_expedited_delivery_correlation_df['preco'].str[1:]).corr(pd.to_numeric(price_expedited_delivery_correlation_df['expedited_orders'])/price_expedited_delivery_correlation_df['total_orders']),4)
print(f"Correlation between price and demand for expedited delivery: {price_expedited_correlation}")

Correlation between price and demand for expedited delivery: 0.0192


Given the 0.0192 correlation between price and probability to demand expedited delivery, we can assume there is effectively no relation between the two in this regard.

In [186]:
# Does the country of origin correlate with product price? 
conn = psycopg2.connect(connection_url)
cursor = conn.cursor()
query = "SELECT * FROM origin_price_correlation"

cursor.execute(query)

rows = []
for item in cursor:
    rows.append(item)

conn.close()

origin_price_correlation_df = pd.DataFrame.from_records(rows, columns=['codigo','produto','preco','ship_country'])
origin_price_correlation_df.head()

codigo                produto   preco ship_country
0  BS001873  Apple Lightning Cable  $21.99           US
1  BS001873  Apple Lightning Cable  $21.99           UK
2  BS001873  Apple Lightning Cable  $21.99           PT
3  BS001873  Apple Lightning Cable  $21.99           GE
4  BS001873  Apple Lightning Cable  $21.99           BR

According to the date, it appears products don't have any price differences if bought from different countries. This could potentially be explained if the delivery fee and taxes are not included in the price.

In [21]:
# Does the country or origin correlate with the ratio of products cancelled?
conn = psycopg2.connect(connection_url)
cursor = conn.cursor()
query = "SELECT * FROM origin_cancellation_ratio"

cursor.execute(query)

rows = []
for item in cursor:
    rows.append(item)

conn.close()

origin_cancellation_ratio_df = pd.DataFrame.from_records(rows, columns=['ship_country','total_orders','cancelled_orders'])
origin_cancellation_ratio_df.head()

ship_country  total_orders  cancelled_orders
0           AU         14325               670
1           BR         14384               769
2           DK         14325               666
3           FR         14325               660
4           GE         14322               660

In [177]:
cancelled_orders_ratio = origin_cancellation_ratio_df['cancelled_orders']/origin_cancellation_ratio_df['total_orders']
data_input = {"Country": origin_cancellation_ratio_df['ship_country'], "Percentage of cancelled orders": round(cancelled_orders_ratio,5)}
cancelled_orders_per_country = pd.DataFrame(data=data_input).sort_values(by=['Percentage of cancelled orders'],ascending=False)
cancelled_orders_per_country

Country  Percentage of cancelled orders
1      BR                         0.05346
0      AU                         0.04677
2      DK                         0.04649
7      UK                         0.04636
6      PT                         0.04622
8      US                         0.04614
4      GE                         0.04608
3      FR                         0.04607
5      JP                         0.04602

With the dataframe displayed above, it's possible to identify that with the exception of Brazil that has a higher order cancellation ratio (5.3%), the other countries of origin have that percentage around 4.6%.

In [178]:
# What countries or origin had more and less of each product sold?
conn = psycopg2.connect(connection_url)
cursor = conn.cursor()
query = "SELECT * FROM origin_products_sold"

cursor.execute(query)

rows = []
for item in cursor:
    rows.append(item)

conn.close()

origin_products_sold_df = pd.DataFrame.from_records(rows, columns=['codigo','produto','ship_country','products_sold'])
origin_products_sold_df.head()

codigo                produto ship_country  products_sold
0  BS001873  Apple Lightning Cable           AU            467
1  BS001873  Apple Lightning Cable           BR            458
2  BS001873  Apple Lightning Cable           DK            451
3  BS001873  Apple Lightning Cable           FR            460
4  BS001873  Apple Lightning Cable           GE            488

In [185]:
print(display(origin_products_sold_df.sort_values(['ship_country','products_sold'], ascending=[True, True]).groupby(['ship_country'], as_index = False, sort = False).nth([0])),display((origin_products_sold_df.sort_values(['ship_country','products_sold'], ascending=[True, False]).groupby(['ship_country'], as_index = False, sort = False).nth([0]))))

codigo                      produto ship_country  products_sold
54  BS001879  Samsung T37F Monitor UHD 37           AU            421
55  BS001879  Samsung T37F Monitor UHD 37           BR            413
56  BS001879  Samsung T37F Monitor UHD 37           DK            378
57  BS001879  Samsung T37F Monitor UHD 37           FR            392
58  BS001879  Samsung T37F Monitor UHD 37           GE            384
59  BS001879  Samsung T37F Monitor UHD 37           JP            397
60  BS001879  Samsung T37F Monitor UHD 37           PT            429
61  BS001879  Samsung T37F Monitor UHD 37           UK            435
62  BS001879  Samsung T37F Monitor UHD 37           US            423

codigo                                            produto ship_country  \
243  BS001900  YABER WiFi Projector Mini Portable Projector 6...           AU   
73   BS001881        TP-Link Tapo Pan/Tilt Smart Security Camera           BR   
74   BS001881        TP-Link Tapo Pan/Tilt Smart Security Camera           DK   
165  BS001891    PHILIPS Sonicare 9900 Prestige Power Toothbrush           FR   
49   BS001878  Logitech MK270R Wireless Keyboard and Mouse Combo           GE   
50   BS001878  Logitech MK270R Wireless Keyboard and Mouse Combo           JP   
258  BS001901                 LG 29WL50S - 29 inch UltraWide UHD           PT   
79   BS001881        TP-Link Tapo Pan/Tilt Smart Security Camera           UK   
170  BS001891    PHILIPS Sonicare 9900 Prestige Power Toothbrush           US   

     products_sold  
243            586  
73             591  
74             577  
165            577  
49             589  
50             586  
258            577  
79             579  
170            593

None None


It's possible to see above the best and worst performing products from each country.